### DISCUSSION TITLE
GitHub action Build job failing

### DISCUSSION THREAD

**Discussion by 10xsqard** [community/community#27341] [Actions]
Team.

I was running the build job but which is failing with below error,  below is my workflow file.
![GitHub_issue](https://user-images.githubusercontent.com/106526567/181304101-3adb85a5-9d6a-448b-8a65-9ab45d28eabd.PNG)
![GitHub_issues](https://user-images.githubusercontent.com/106526567/181304115-820471f2-6fdd-4829-a3e1-69d517d52b6a.PNG)

name: CI

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "master" branch
  push:
    branches: [ "QA" ]
  pull_request:
    branches: [ "QA" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v3

      # Runs a single command using the runners shell
      - name: Set version
        run: echo "PACKAGE_VERSION=$(date +'%Y.%m.%d').$GITHUB_RUN_NUMBER" >> $GITHUB_ENV

      # Runs a set of commands using the runners shell
      - name: Run a multi-line script
        run: |
          echo Add other actions to build,
          echo test, and deploy your project.
          
      - name: Setup .NET Core
        uses: actions/setup-dotnet@v1.6.0
        with:
         dotnet-version: ${{ env.DOTNET_SDK_VERSION }}
         
      - name: Install Wyam
        run: dotnet tool install --tool-path . wyam.tool
        
      - name: Setup dotnet
        uses: actions/setup-dotnet@v2
        with: 
          dotnet-version: '2.1.x'
              
      #- name: Generate site
        #run: ./wyam build -r blog -t Stellar src
        
      - name: Install Octopus CLI
        run: |
         sudo apt update && sudo apt install --no-install-recommends gnupg curl ca-certificates apt-transport-https && \
         curl -sSfL https://apt.octopus.com/public.key | sudo apt-key add - && \
         sudo sh -c "echo deb https://apt.octopus.com/ stable main > /etc/apt/sources.list.d/octopus.com.list" && \
         sudo apt update && sudo apt install octopuscli
           
         - name: Package blog.rousseau.dev
         run: |
         mkdir -p ./packages/blog.rousseau.dev/src/
         cp .firebaserc firebase.json ./packages/blog.rousseau.dev/
         cp -avr ./src/output ./packages/blog.rousseau.dev/src/
         octo pack --id="blog.rousseau.dev" --format="Zip" --version="${{ env.PACKAGE_VERSION }}" --basePath="./packages/blog.rousseau.dev" --outFolder="./packages"

          - name: Push blog.rousseau.dev to Octopus
             run: octo push --package="./packages/blog.rousseau.dev.${{ env.PACKAGE_VERSION }}.zip" --server="${{ secrets.OCTOPUS_SERVER_URL }}" --apiKey="${{ secrets.OCTOPUS_API_KEY }}"

**Comment by irongut** [comment]
The 127 error code indicates “command not found”. Have you tried running that series of commands locally to see if they work?

That said, you're trying to install Octopus CLI so you would be better using this Action rather than shell commands: https://github.com/marketplace/actions/install-octopus-cli

**Reply by 10xsqard** [reply]
HI Team.

Thanks for your time on this.
I have corrected indentation of install OCtopus CLI and it got success now and now I have a issue in steps to copy and push packages to octopus and generate site, Would you able to help us for better line of code, here is my code in yaml file where i'm using it

![image](https://user-images.githubusercontent.com/106526567/182077411-70ace9dd-5a53-4da6-84b8-76cc85b30cad.png)
![Github_packagepush](https://user-images.githubusercontent.com/106526567/182077471-d6fea3d6-f2ff-4733-8287-fa22b1b2a156.PNG)

- name: Package blog.rousseau.dev
        run: |
          mkdir -p ./packages/blog.rousseau.dev/src/
          cp .firebaserc firebase.json ./packages/blog.rousseau.dev/
          cp -avr ./src/output ./packages/blog.rousseau.dev/src/
          octo pack --id="blog.rousseau.dev" --format="Zip" --version="${{ env.PACKAGE_VERSION }}" --basePath="./packages/blog.rousseau.dev" --outFolder="./packages"

          - name: Push blog.rousseau.dev to Octopus
             run: octo push --package="./packages/blog.rousseau.dev.${{ env.PACKAGE_VERSION }}.zip" --server="${{ secrets.OCTOPUS_SERVER_URL }}" --apiKey="${{ secrets.OCTOPUS_API_KEY }}"


 #- name: Generate site
        #run: ./wyam build -r blog -t Stellar src

**Reply by irongut** [reply]
I'm not an Octopus user but generally I look for a custom Action on the Marketplace first and use shell commands if I can't find one. There are quite a few Octopus actions available: https://github.com/marketplace?type=actions&query=octopus+

I think this might be the one you're looking for: https://github.com/marketplace/actions/push-package-to-octopus-deploy

**Reply by 10xsqard** [reply]
HI Thank you

I also using the same from market place to make packages and push packages to octopuss but its failing in  GITHub job 
Here is the code in workflow file

- name: Install Octopus CLI 🐙
        uses: OctopusDeploy/install-octopus-cli-action@v1
        with:
         version: '*'
      
      - name: Push a package to Octopus Deploy 🐙
        uses: OctopusDeploy/push-package-action@v2
        env:
         OCTOPUS_API_KEY: ${{ secrets.API_KEY }}
         OCTOPUS_HOST: ${{ secrets.SERVER }}
         OCTOPUS_SPACE: 'Spaces-1'
        with:
          packages: |
           package1.zip
           packages/**/*.zip

![image](https://user-images.githubusercontent.com/106526567/182212588-1785d958-b81b-4dad-9904-fbc6096154e9.png)

please help

**Comment by airtower-luna** [comment]
You seem to have broken indentation here:

```yaml
  - name: Install Octopus CLI
    run: |
     sudo apt update && sudo apt install --no-install-recommends gnupg curl ca-certificates apt-transport-https && \
     curl -sSfL https://apt.octopus.com/public.key | sudo apt-key add - && \
     sudo sh -c "echo deb https://apt.octopus.com/ stable main > /etc/apt/sources.list.d/octopus.com.list" && \
     sudo apt update && sudo apt install octopuscli
       
     - name: Package blog.rousseau.dev
     run: |
```

The `- name: Package blog.rousseau.dev` line is indented at the same level as the `sudo` command before, so it (and everything after, because that's indented at least as much) is still considered part of the `run` block. And `- name: [...]` is not a valid command. Make sure each item in your `steps` is indented at the same levels.
